records/generator.go

```diff
@@ -57,8 +67,8 @@ func (rg *RecordGenerator) ParseState(c Config, masters ...string) error {
 
 // Tries each master and looks for the leader
 // if no leader responds it errors
-func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
-	var sj state.State
+func (rg *RecordGenerator) findMaster(masters ...string) (*state.State, error) {
```

Comment:
returning error objects upon an error condition is idiomatic Go. if err == nil then there's no error. having two possible error states to check (error object, or state == nil) is confusing and not intuitive.